cou/apps/auxiliary.py

```diff
@@ -204,10 +205,55 @@ class MysqlInnodbCluster(AuxiliaryApplication):
     wait_timeout = LONG_IDLE_TIMEOUT
 
 
-# NOTE (gabrielcocenza): Although CephOSD class is empty now, it will be
-# necessary to add post upgrade plan to set require-osd-release. Registering on
-# AuxiliaryApplication can be easily forgot and ceph-osd can't be instantiated
-# as a normal OpenStackApplication.
 @AppFactory.register_application(["ceph-osd"])
 class CephOSD(AuxiliaryApplication):
     """Application for ceph-osd."""
+
+    def pre_upgrade_steps(
+        self, target: OpenStackRelease, units: Optional[list[COUUnit]]
```

Comment:
```suggestion
        self, target: OpenStackRelease, units: Optional[list[COUUnit]] = None
```

Reply:
Base has no default value, so why adding it here?

Reply:
Because ceph-osd it's not suppose to receive units, but this task should be covered on the `upgrade_plan_sanity_checks`, so I'm ok leaving this way. Solving it.